packages/dapp/src/components/Article.tsx

```diff
@@ -0,0 +1,68 @@
+import * as React from "react";
+import { CivilEditor, plugins } from "@joincivil/editor";
+import { getCivil } from "../helpers/civilInstance";
+
+export interface ArticleState {
+  value: any | undefined;
+  error: string;
+}
+
+export interface ArticleProps {
+  match: any;
+}
+
+class Article extends React.Component<ArticleProps, ArticleState> {
+  constructor(props: ArticleProps) {
+    super(props);
+    this.state = {
+      value: undefined,
+      error: "",
+    };
+  }
+
+  public async componentWillMount(): Promise<void> {
+    return this.initArticle();
+  }
+
+  public render(): JSX.Element {
+    return (
+      <>
+        {this.state.value && (
+          <CivilEditor plugins={plugins} value={this.state.value} onChange={this.onChange} readOnly={true} />
+        )}
+        {!this.state.value && "loading..."}
+      </>
+    );
+  }
+
+  private onChange = ({ value }: { value: any }): void => {
+    // gave warning (treated as error on production build) if function left empty
+    return;
+  };
+
+  private initArticle = async () => {
+    const newsroom = await this.getNewsroom();
+    if (newsroom) {
+      const article = await newsroom.loadArticle(this.props.match.params.articleId);
```

Comment:
i think ultimately the newsroom class needs to do less, it should just return the uri and your app needs to deal with it